### PR TITLE
Fixed issue #16187: No proper error message when copying survey and ID already exists

### PR DESCRIPTION
**Pull request by gabrieljenik**
Added warning about "custom survey ID is taken"
Reviewed help text of the custom survey id field

### CODE REVIEW

File: application/helpers/admin/import_helper.php [hunk 1]
```diff
@@ -1285,6 +1285,9 @@ function XMLImportSurvey($sFullFilePath, $sXMLdata = null, $sNewSurveyName = nul
         if ($newSurvey->sid) {
             $iNewSID = $results['newsid'] = $newSurvey->sid;
             $results['surveys']++;
+            if (!empty($iDesiredSurveyId) && $iNewSID != $iDesiredSurveyId) {
+                $results['importwarnings'][] = gT("The desired survey ID was invalid and could not be assigned to the new survey.");
```
Comment (Shnoulle) [inline application/helpers/admin/import_helper.php:1289]: clear enough

File: application/views/surveyAdministration/tabCopy_view.php [hunk 2]
```diff
@@ -42,10 +42,10 @@
                         <div class="">
                             <input type='number' step="1" min="1" max="999999" id='copysurveyid' size='82' name='copysurveyid' value='' class="form-control" />
                         </div>
-                        <div class="">
-                          <p class="form-control-static">
+                        <div class="help-block">
                             <span class='annotation text-info'><?php echo  gT("Optional"); ?> </span>
-                          </p>
+                            -
+                            <?= gT("If ID is taken, a random one will be assigned."); ?> </span>
```
[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/views/surveyAdministration/tabCopy_view.php:48]: Maybe«If survey id is taken …» since we use "New survey id:" for label.

Microfix

**Comment by c-schmitz** [comment]
Fixed the PR wording - please use the same for the commit.

File: application/views/surveyAdministration/tabCopy_view.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (c-schmitz) [reply application/views/surveyAdministration/tabCopy_view.php:48]: "If the new survey ID is already used, a random one will be assigned."

File: application/helpers/admin/import_helper.php (hunk 1 above)
Comment (c-schmitz) [reply application/helpers/admin/import_helper.php:1289]: "The desired survey ID was already in use, therefore a random one was assigned."

**Review by c-schmitz — CHANGES_REQUESTED** [review]

**Comment by Shnoulle** [comment]
@c-schmitz when reviewing, when there are wording : did i set you as reviewer each time ? Here: I hesitated to ask you for the review.

**Comment by gabrieljenik** [comment]
@c-schmitz Changes done.

**Comment by olleharstedt** [comment]
Thanks!

**Pull request merged**